src/entities/FormattingNode/index.ts

```diff
@@ -1,46 +1,258 @@
-import { TextNode } from '../TextNode';
-import { FormattingNodeConstructorParameters, FormattingNodeName, createFormattingNodeName } from './types';
+import {
+  FormattingNodeConstructorParameters,
+  InlineToolName,
+  InlineToolData
+} from './types';
+import { ChildNode, InlineFragment, InlineNode, InlineNodeSerialized, ParentNode } from '../interfaces';
+
+export * from './types';
+
+export interface FormattingNode extends ChildNode, ParentNode {
+  children: (InlineNode & ChildNode)[]
+}
 
 /**
  * FormattingNode class represents a node in a tree-like structure, used to store and manipulate formatted text content
  */
-export class FormattingNode {
+@ParentNode
+@ChildNode
+export class FormattingNode implements InlineNode {
   /**
    * Private field representing the name of the formatting tool applied to the content
    */
-  #name: FormattingNodeName;
+  #tool: InlineToolName;
+
+  /**
+   * Private field representing any additional data associated with the formatting tool
+   */
+  #data?: InlineToolData;
 
   /**
-   * Private field representing any additional data associated with the formatting
+   * Constructor for FormattingNode class.
+   *
+   * @param args - FormattingNode constructor arguments.
+   * @param args.tool - The name of the formatting tool applied to the content.
+   * @param args.data - Any additional data associated with the formatting.
    */
-  #data: Record<string, unknown>;
+  constructor({ tool, data }: FormattingNodeConstructorParameters) {
+    this.#tool = tool;
+    this.#data = data;
+  }
 
   /**
-   * Private field representing the length of the text content stored as a children in this node
+   * Returns text value length of current node (including subtree)
    */
-  #length: number;
+  public get length(): number {
+    return this.children.reduce((sum, child) => sum + child.length, 0);
+  }
 
   /**
-   * Private array field of FormattingNode and TextNode objects, representing any child nodes of this node
+   * Returns serialized value of the node: text and formatting fragments
    */
-  #children: (FormattingNode | TextNode)[];
+  public get serialized(): InlineNodeSerialized {
+    return {
+      text: this.getText(),
+      fragments: this.getFragments(),
```

Comment:
Will all text fragments be included here or only those that contain formatting?

Reply:
`text` prop will contain the whole text value, `fragments` will contain all inline formatting in following format:
```
interface InlineFragment {
  tool: InlineToolName;
  data?: InlineToolData;
  range: [start: number, end: number];
}
```